### PR TITLE
Update CI configuration

### PR DESCRIPTION
**Pull request by IvanSavenko**
Fixes for CI issues discovered after Android Qt launcher release.  Not insisting on these changes, or on their form, so any better solutions for these issues are welcomed.

Changes:
- Remove bundle_release job - AAB's generated by Qt+CMake are currently broken. If fixed - job can be restored via git log / git revert. Also removed JNI upload step since it is only needed for unified AAB generation.
- Upload generated AAB file in addition to APK for Android CI runs
- Build signed Android build only when running on master branch. Previously this was part of bundle_release job, and as side effect - apk's generated on master branch were generated as "VCMI daily" builds, and not release builds.
- Minor cleanup: disable not used "Additional logs" step from the matrix
- Minor cleanup: move code validation into a separate job outside of the matrix. It does not depends on builds and is only run in a single CI instance

Missing change: 
- generate 32-bit and 64-bit aab's with different build ID's so both Android builds from same CI run can be uploaded to Google Play. Not sure how to do this or how to do this nicely in our CI. Help would be welcomed.

### CODE REVIEW

File: .github/workflows/github.yml [hunk 1]
```diff
@@ -437,3 +328,31 @@ jobs:
...
+        - name: Ensure LF line endings
+          run: |
+            find . -path ./.git -prune -o -path ./AI/FuzzyLite -prune -o -path ./test/googletest \
+            -o -path ./osx  -prune -o -type f \
+            -not -name '*.png' -and -not -name '*.vcxproj*' -and -not -name '*.props' -and -not -name '*.wav' -and -not -name '*.webm' -and -not -name '*.ico' -and -not -name '*.bat' -print0 | \
+            { ! xargs -0 grep -l -z -P '\r\n'; }
+
+        - name: Validate JSON
+          # the Python yaml module doesn't seem to work on mac-arm
+          # also, running it on multiple presets is redundant and slightly increases already long CI built times
```
Comment (Alexander-Wilms) [inline .github/workflows/github.yml:355]:
```suggestion
```

File: .github/workflows/github.yml [hunk 2]
```diff
@@ -250,11 +237,11 @@ jobs:
...
-    - name: Additional logs
-      if: ${{ failure() && steps.cpack.outcome == 'failure' && matrix.platform == 'msvc' }}
-      run: |
-        cat '${{github.workspace}}/out/build/${{matrix.preset}}/_CPack_Packages/win32/NSIS/project.nsi'
-        cat '${{github.workspace}}/out/build/${{matrix.preset}}/_CPack_Packages/win32/NSIS/NSISOutput.log'
+#    - name: Additional logs
+#      if: ${{ failure() && steps.cpack.outcome == 'failure' && matrix.platform == 'msvc' }}
+#      run: |
+#        cat '${{github.workspace}}/out/build/${{matrix.preset}}/_CPack_Packages/win32/NSIS/project.nsi'
+#        cat '${{github.workspace}}/out/build/${{matrix.preset}}/_CPack_Packages/win32/NSIS/NSISOutput.log'
```
Comment (kambala-decapitator) [inline .github/workflows/github.yml:244]: I'd simply remove this

**Review by kambala-decapitator — APPROVED** [review]
Can be merged, but it's still not a final solution for Android

**Comment by kambala-decapitator** [comment]
BTW why merging to master? Better target beta

**Comment by IvanSavenko** [comment]
> Can be merged, but it's still not a final solution for Android

Yeah, I know. This is mostly to have something that is not completely broken.

> BTW why merging to master? Better target beta

Changed.

**Pull request merged**